inc/class-s3-uploads.php

```diff
@@ -162,6 +174,54 @@ public function get_original_upload_dir() {
 		return $this->original_upload_dir;
 	}
 
+	/**
+	 * Reverse a file url in the uploads directory to the params needed for S3.
+	 *
+	 * @param string $url
+	 * @return array
+	 */
+	public function get_s3_location_for_url( string $url ) : ?array {
+		$s3_url = 'https://' . $this->get_s3_bucket() . '.s3.amazonaws.com/';
+		if ( strpos( $url, $s3_url ) === 0 ) {
+			$parsed = parse_url( $url );
+			return [
+				'bucket' => $this->get_s3_bucket(),
+				'key'    => ltrim( $parsed['path'], '/' ),
+				'query'  => $parsed['query'] ?? null,
+			];
+		}
+		$upload_dir = wp_upload_dir();
+
+		if ( strpos( $url, $upload_dir['baseurl'] ) === false ) {
+			return null;
+		}
+
+		$path = str_replace( $upload_dir['baseurl'], $upload_dir['basedir'], $url );
+		$parsed = parse_url( $path );
+		return [
+			'bucket' => $parsed['host'],
+			'key'    => ltrim( $parsed['path'], '/' ),
+			'query'  => $parsed['query'] ?? null,
+		];
+	}
+
+	/**
+	 * Reverse a file path in the uploads directory to the params needed for S3.
+	 *
+	 * @param string $url
+	 * @return array
+	 */
+	public function get_s3_location_for_path( string $path ) : ?array {
+		$parsed = parse_url( $path );
```

Comment:
Use `wp_parse_url` instead. This is also flagged by hm-linter, So it will fix that as well.

```suggestion
		$parsed = wp_parse_url( $path );
```

Reply:
I fixed this in #401 , so think I'll just leave the commit there.